VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitWebviewFragment.java

```diff
@@ -101,56 +112,82 @@ public void onActivityCreated(Bundle savedInstanceState) {
         webView.getSettings().setJavaScriptEnabled(true);
         URLInterceptorWebViewClient client =
                 new URLInterceptorWebViewClient(getActivity(), webView) {
-            @Override
-            public void onReceivedError(WebView view, int errorCode,
-                    String description, String failingUrl) {
-                hideLoadingProgress();
-                pageIsLoaded = false;
-                ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, false);
-            }
-
-            // TODO: Restore these annotations when we upgrade our compile SDK version to Marshmallow
-            //@Override
-            //@TargetApi(Build.VERSION_CODES.M)
-            @TargetApi(23)
-            public void onReceivedHttpError(WebView view, WebResourceRequest request,
-                    WebResourceResponse errorResponse) {
-                switch (errorResponse.getStatusCode()) {
-                    case HttpURLConnection.HTTP_FORBIDDEN:
-                    case HttpURLConnection.HTTP_UNAUTHORIZED:
-                    case HttpURLConnection.HTTP_NOT_FOUND:
-                        EdxCookieManager.getSharedInstance().tryToRefreshSessionCookie();
-                        break;
-                }
-            }
-
-            public void onPageFinished(WebView view, String url) {
-                if (url != null && url.equals("data:text/html," + EMPTY_HTML)) {
-                    //we load a local empty html page to release the memory
-                } else {
-                    pageIsLoaded = true;
-                }
-
-                //TODO -disable it for now. as it causes some issues for assessment
-                //webview to fit in the screen. But we still need it to show additional
-                //compenent below the webview in the future?
-                // view.loadUrl("javascript:EdxAssessmentView.resize(document.body.getBoundingClientRect().height)");
-                ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, true);
-                hideLoadingProgress();
-            }
-        };
+                    private boolean didReceiveError = false;
+
+                    @Override
+                    public void onReceivedError(WebView view, int errorCode,
+                                                String description, String failingUrl) {
+                        didReceiveError = true;
+                        hideLoadingProgress();
+                        pageIsLoaded = false;
+                        ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, false);
+                        setFragmentState(true);
```

Comment:
If we're displaying loading errors, then we should have a different error message than the "no connectivity" one. Maybe we can use `network_service_unavailable` as a generic error message? This should perhaps also be done from the `onReceivedHttpError()` callback (maybe with a slightly different message appropriate for receiving HTTP errors?).

Note that you'll need to change the `setFragmentState()` method (which already is not best named) into something more generic that can take error messages and icons as parameters. Since the icon will be set dynamically, you can also reduce your `no_connectivity_layout` layout to just a `TextView` with the `content_unavailable_error_text` applied, on which you can then set the relevant `IconDrawable` as the top compound drawable. The `no_connectivity_layout` layout can in that case be deleted, since the `TextView` can be defined in 2-3 lines.